### PR TITLE
All unit tests passing on JRuby

### PR DESCRIPTION
**Pull request by lucasallan**

### CODE REVIEW

**Comment by jodosha** [comment]
@lucasallan Thank you, this would be a huge milestone for the framework! 👍

**Comment by joneslee85** [comment]
@jodosha welcome back

File: test/integration/cdn/test_application.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@
     @hanami_env       = ENV['HANAMI_ENV']
     ENV['HANAMI_ENV'] = 'production'
 
-    assert system("HANAMI_ENV=production bundle exec hanami assets precompile")
+    system("HANAMI_ENV=production bundle exec hanami assets precompile")
```
Comment (jodosha) [inline test/integration/cdn/test_application.rb]: Here we're asserting that the returning value of `Kernel.system` is `true`. This specific returning value means that the system call is successful and so we can tell that the assets precompilation went good.

Please restore this assertion. 😄

Comment (lucasallan) [inline test/integration/cdn/test_application.rb]: This is a working in progress, I'll review everything before I submit for review.

In this specific case, looks like the command is successfully executed but the return is nil. I'm currently investigating it and it might be a fix I have to do in JRuby side.

Thanks for bringing this up.

**Comment by lucasallan** [comment]
@jodosha It's ready for reviewing. I'm keeping jruby tests in the `allowed_failures` on TravisCI, just because we have a couple of tests that are occasionally failing - I'm currently digging into it.

File: test/integration/cdn/test_application.rb [hunk 2]
```diff
@@ -9,8 +9,7 @@
 
     @hanami_env       = ENV['HANAMI_ENV']
     ENV['HANAMI_ENV'] = 'production'
-
-    assert system("HANAMI_ENV=production bundle exec hanami assets precompile")
+    assert system("bundle exec hanami assets precompile")
```
Comment (jodosha) [inline test/integration/cdn/test_application.rb]: @lucasallan Why did you removed `HANAMI_ENV=production`? This is the closest simulation of a deployment scenario.

**Comment by jodosha** [comment]
@lucasallan Hey thanks! 💚 

I noticed that all the generated `Gemfile` lack of the following lines:

```ruby
group :test, :development do
   gem 'dotenv', '~> 2.0'
 end
```

File: test/integration/cdn/test_application.rb (hunk 2 above)
Comment (lucasallan) [inline test/integration/cdn/test_application.rb]: We're already setting the `HANAMI_ENV` to `production` - line 11.

**Comment by lucasallan** [comment]
@jodosha I'll fix the Gemfile, if you guys have any other issue with JRuby - you can ping me and I'll work on it.

**Comment by jodosha** [comment]
@lucasallan Thank you very much 💚

**Comment by lucasallan** [comment]
Thank you!! @jodosha

File: test/integration/cdn/test_application.rb (hunk 2 above)
Comment (lucasallan) [inline test/integration/cdn/test_application.rb]: There is an issue with JRuby when executing with `HANAMI_ENV=<anything>`

```
[1] pry(#<CDN (Application)>)> system("bundle exec hanami assets precompile")
=> true
[2] pry(#<CDN (Application)>)> system("HANAMI_ENV=production bundle exec hanami assets precompile")
=> nil
[3] pry(#<CDN (Application)>)> system("HANAMI_ENV=test bundle exec hanami assets precompile")
=> nil
[4] pry(#<CDN (Application)>)> system("HANAMI_ENV=development bundle exec hanami assets precompile")
=> nil
[5] pry(#<CDN (Application)>)> system("echo $HANAMI_ENV")
production
=> true
[6] pry(#<CDN (Application)>)> system("HANAMI_ENV=production ls -la")
=> nil
[7] pry(#<CDN (Application)>)> system("ls -la")
total 72
drwxr-xr-x  17 lucasamorim  staff   578 Aug 10 11:51 .
drwxr-xr-x   4 lucasamorim  staff   136 Jul 28 10:25 ..
-rw-r--r--   1 lucasamorim  staff    23 Jul 28 10:25 .env
-rw-r--r--   1 lucasamorim  staff   223 Jul 28 10:25 .env.development
-rw-r--r--   1 lucasamorim  staff   209 Jul 28 10:25 .env.test
-rw-r--r--   1 lucasamorim  staff    62 Jul 28 10:25 .gitignore
-rw-r--r--   1 lucasamorim  staff    44 Jul 28 10:25 .hanamirc
-rw-r--r--   1 lucasamorim  staff   998 Jul 28 10:25 Gemfile
-rw-r--r--   1 lucasamorim  staff  4014 Aug 10 11:51 Gemfile.lock
-rw-r--r--   1 lucasamorim  staff   164 Jul 28 10:25 Rakefile
drwxr-xr-x   6 lucasamorim  staff   204 Jul 28 10:25 app
drwxr-xr-x   6 lucasamorim  staff   204 Jul 28 10:25 config
-rw-r--r--   1 lucasamorim  staff    60 Jul 28 10:25 config.ru
drwxr-xr-x   3 lucasamorim  staff   102 Jul 28 10:25 db
drwxr-xr-x   5 lucasamorim  staff   170 Jul 28 10:25 lib
drwxr-xr-x   4 lucasamorim  staff   136 Aug 12 17:16 public
drwxr-xr-x   9 lucasamorim  staff   306 Jul 28 10:25 spec
=> true
```

It does not seem to be related to Hanami, since it doesn't work when I run `system("HANAMI_ENV=production ls -la")` See logs above - I'm currently investigating it on the JRuby side.

**Comment by jodosha** [comment]
@lucasallan I've just merged this, thanks! I may need your help in the near future for a WIP branch.
